src/App.jsx

```diff
@@ -13,34 +15,85 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const sortList = (goods, sortBy, isReversed) => {
+  const sortedGoods = [...goods].sort((a, b) => {
+    switch (sortBy) {
+      case 'alphabetically':
+        return a.localeCompare(b);
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+      case 'length':
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return sortedGoods;
+};
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortedBy, setSortedBy] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const sortedGoods = sortList(goods, sortedBy, reversed);
+  const resetGoods = () => {
+    setGoods(goodsFromServer);
+    setSortedBy('');
+    setReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortedBy !== 'alphabetically',
+          })}
+          onClick={() => setSortedBy('alphabetically')}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortedBy !== 'length',
```

Comment:
make variables for sortField and use in all places. for example it is can be object.

const SORT_FIELD = {
   None: '',
   Length: 'length',
   Alphabetically: 'alphabetically'
}
```suggestion
            'is-light': sortedBy !== SORT_FIELD.Length,
```